### PR TITLE
duplicate comment removal fix and tests for issue #1490

### PR DESCRIPTION
**Pull request by ThLeu**
tests for issue #1490 

It seems that this issue is fixed already, the tests work without a problem.

If you do not like my coding style for tests (mainly the duplication of test code and the AAA (Arrange Act Assert) style) please tell me and I will change that 😄 . Although I recommend that style 😛 for several reasons.

### CODE REVIEW

**Review by ftomassetti — APPROVED** [review]
+1

File: javaparser-core-testing/src/test/java/com/github/javaparser/ast/comments/CommentTest.java [hunk 1]
```diff
@@ -69,4 +77,125 @@ public void unicodeEscapesArePreservedInComments() {
...
+                EOL);
+
+        MethodDeclaration methodDeclaration = cu.findFirst(MethodDeclaration.class).get();
+
+        // Act
+        Javadoc javadoc = new Javadoc(JavadocDescription.parseText("Change Javadoc"));
+        methodDeclaration.setJavadocComment("", javadoc);
+
+        // Assert
+        assertEquals("public class MyClass {" + EOL +
```
Comment (matozoid) [inline javaparser-core-testing/src/test/java/com/github/javaparser/ast/comments/CommentTest.java]: Take a look at assertEqualsNoEol (or so) - building on unix and windows is painful, and has everything to do with comparing eol's :-)

Comment (ThLeu) [reply javaparser-core-testing/src/test/java/com/github/javaparser/ast/comments/CommentTest.java]: To be honest, I don't see much of a difference between using EOL and \n everywhere :)
But I guess assertEqualsNoEol gives the freedom to copy pase in Intellij IDEA, is that why you use it?
I will change it if you like.

**Comment by ThLeu** [comment]
Ok, As i guessed, the bug isn't fixed and just happens when using the LexicalPreservingPrinter.
My last commits will fix the issue #1490 for javadocs BUT....
1. we have the same indentation problems here as we had / have in the Difference class of lexical preservation. It keeps too many spaces for now.
2. normal comments are not supported yet (as they were before), but this is a different issue.

I added unit tests that are ignored for now (with the reason why they are ignored).
So this PR could be merged if you like and the other issues can be handled separately.

fixes #1490

**Comment by ThLeu** [comment]
As you can see in the commits I added the Range to the comparison of the TokenTextElements if there is more than one. 
The same problem happened afterwards with the NodeText.findElement method which uses a TextElementMatcher to find the correct element. However the default behaviour always found the first comment of the two, even when changing the second one. I refined that by enabling the combination of different TextElementMatchers with the `and()` Method on the interface and adding a first one `matchByRange()` to TextElement.

File: javaparser-core-testing/src/test/java/com/github/javaparser/ast/comments/CommentTest.java (hunk 1 above)
Comment (matozoid) [reply javaparser-core-testing/src/test/java/com/github/javaparser/ast/comments/CommentTest.java]: Do change it - EOL is the *platform specific* end of line character, and `assertEqualsNoEol` replaces each and every eol character to \n so you can safely compare to strings with \n in it. These tests fail on Windows.

Comment (ThLeu) [reply javaparser-core-testing/src/test/java/com/github/javaparser/ast/comments/CommentTest.java]: I changed it now. Maybe I misunderstand something but I can safely say that the test was running on Windows before, as I programmed it on Windows.
Also `Utils.EOL` already uses `System.getProperty("line.separator")` which works on all platforms and `assertEqualsNoEol`, as I understand it, basically replaces the "\n" back to `Utils.EOL`, which is why I didn't really see a difference between before and after, maybe I am missing something.
Anyway, I changed it and I didn't really want to start a big discussion :)

**Comment by matozoid** [comment]
Thanks @ThLeu !

**Pull request merged**